ml-agents/mlagents/trainers/sac/trainer.py

```diff
@@ -149,15 +149,16 @@ def _process_trajectory(self, trajectory: Trajectory) -> None:
             agent_buffer_trajectory["environment_rewards"]
         )
         for name, reward_signal in self.optimizer.reward_signals.items():
-            if isinstance(reward_signal, RewardSignal):
-                evaluate_result = reward_signal.evaluate_batch(
-                    agent_buffer_trajectory
-                ).scaled_reward
-            else:
+            if isinstance(reward_signal, BaseRewardProvider):
```

Comment:
Same comments about tf vs torch here and in _process_trajectory

Reply:
Added comments throughout